packages/web-components/fast-components/src/radio-group/radio-group.ts

```diff
@@ -0,0 +1,254 @@
+            this.slottedRadioButtons.forEach((item: any) => {
+                if (item instanceof HTMLElement) {
+                    radioButtons.push(item as HTMLInputElement);
+                }
+            });
+        }
+        return radioButtons;
+    };
+
+    private keypressHandler = (e: KeyboardEvent): void => {
+        const radio: HTMLInputElement | null = e.target as HTMLInputElement;
+        if (radio) {
+            radio.setAttribute("tabindex", radio.checked ? "0" : "-1");
+        }
+    };
+
+    private handleRadioChange = (e: CustomEvent): void => {
+        const changedRadio: HTMLInputElement = e.target as HTMLInputElement;
+        if (changedRadio.checked) {
+            this.getFilteredRadioButtons().forEach((radio: HTMLInputElement) => {
+                if (radio !== changedRadio) {
+                    radio.checked = false;
+                    radio.setAttribute("tabindex", "-1");
+                }
+            });
+            this.selectedRadio = changedRadio;
+            this.value = changedRadio.value;
+        }
+    };
+
+    private moveToRadioByIndex = (
+        group: RadioControl[],
+        index: number,
+        inToolbar: boolean
+    ) => {
+        const radio: RadioControl = group[index];
+        if (!radio.readOnly && !inToolbar) {
+            radio.checked = true;
+            radio.setAttribute("tabindex", "0");
+        } else {
+            radio.setAttribute("tabindex", "-1");
+        }
+        this.selectedRadio = radio;
+        radio.focus();
+    };
+
+    private moveRightOffGroup = () => {
+        this.selectedRadio = null;
+        (this.nextElementSibling as HTMLInputElement).focus();
+    };
+
+    private moveLeftOffGroup = () => {
+        this.selectedRadio = null;
+        (this.previousElementSibling as HTMLInputElement).focus();
+    };
+
+    /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+    private handleFocusOut = (e: FocusEvent) => {
+        const group: RadioControl[] = this.getFilteredRadioButtons();
+        if (!this.selectedRadio) {
+            group[0].setAttribute("tabindex", "0");
+        }
+    };
+
+    private clickHandler = (e: MouseEvent): void => {
+        const radio: HTMLInputElement | null = e.target as HTMLInputElement;
+        if (radio) {
+            const group: RadioControl[] = this.getFilteredRadioButtons();
+            if (radio.checked || group.indexOf(radio) === 0) {
+                radio.setAttribute("tabindex", "0");
+                this.selectedRadio = radio;
+            } else {
+                radio.setAttribute("tabindex", "-1");
+                this.selectedRadio = null;
+            }
+        }
+    };
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        const group: RadioControl[] = this.getFilteredRadioButtons();
+        let index: number = 0;
+        const toolbar: HTMLElement | null | undefined = this.parentElement?.closest(
+            '[role="toolbar"]'
+        );
```

Comment:
Are there other circumstances where the radio group could be hosted in a container that also uses key navigation that we need to consider (ie. data grids, menus, etc...)?

Reply:
("key navigation" = using arrow keys)

Reply:
I'm going by https://w3c.github.io/aria-practices/#for-radio-group-contained-in-a-toolbar which only indicated a special scenario for toolbar, otherwise I think it is supposed to use the pattern described for Not in a toolbar

Reply:
+1 to the above - currently the only delta in implementation from a keyboarding standpoint is the one called out for toolbar in the spec